Compatibility tweaks for Windows and MSYS2

LAGonauta opened the pull request:
Changes:
- ~~Fixes inability to build in MSYS2. Some Windows-native libraries had to be blacklisted as the CMake script is not able to find them. This is not necessary when cross-compiling from Linux.~~
- ~~Use `ShellExecuteA` instead of `AppInfo.launch_default_for_uri`. Allows Dino to launch UWP applications, can be removed after https://gitlab.gnome.org/GNOME/glib/-/issues/1991 is closed.~~
- Set file as downloaded from the internet so Windows is able to prompt the user to be sure they wants to open a file
- Get file mime-type from the extension on Windows. `FileInfo.get_content_type()` does not return correct values as it doesn't look inside the file. Fixes image preview.
- Increase font size on Windows a little. This normalizes the default appearence on Linux to Windows and makes it more readable.
- Added executable information and icon on Windows
- Added default PGPE path for PGP support

~~Notification currently works by flashing the taskbar entry, but should be complete when the `GNotification` backend is ready. Right now I do not think there is anything else missing for full feature parity.~~

Notifications are implemented fully as a plugin, this might eventually be replaced by a GNotification backend.

How to build on MSYS2:
https://github.com/dino/dino/issues/309#issuecomment-630493660

Any suggestion is welcome 🙂
Thanks!

LAGonauta committed:
Windows compatibility Tweaks

LAGonauta committed:
fix build on newest MSYS2

LAGonauta committed:
Do not search for the built-in libraries when compiling with MINGW

LAGonauta committed:
Added _WIN32 define to VALAC on Windows

LAGonauta committed:
Add missing _WIN32

LAGonauta committed:
Add support for OpenPGP on Windows

LAGonauta committed:
Use ShellExecute instead of AppInfo to open files on Windows

LAGonauta committed:
Use slight larger font on Windows so it matches Linux style

Also fixes some fuzzy fonts.

LAGonauta committed:
Fixed some Windows not appearing when opening file

LAGonauta committed:
Set alternate file stream for downloaded files.

LAGonauta committed:
Added information and Dino icon to Windows executable

mar-v-in commented on cmake/PkgConfigWithFallback.cmake:
```diff
@@ -13,11 +13,16 @@ function(find_pkg_config_with_fallback name)
         set(${name}_FOUND ${${name}_PKG_CONFIG_FOUND})
 
+        if(MINGW)
+            list(FILTER ${name}_PKG_CONFIG_LIBRARIES EXCLUDE REGEX "(ole32)|(gdi32)|(imm32)|(shell32)|(winmm)|(dwmapi)|(setupapi)|(cfgmgr32)|(ws2_32)")
```
This seems to be a hack specific to your build system. These libraries look like they actually are linker dependencies and thus should be included.

LAGonauta replied:
Yes, it is a quirk of MSYS2. It is not needed when cross-compiling from Fedora.
Why does it need to manually find each DLL, by the way? Doesn't the linker know where they are anyway? 🤔

---

I changed the blacklist to explicity add the implicit library directory to the hint path, is this OK? Is there a better way?

mar-v-in commented on libdino/src/application.vala:
```diff
@@ -58,6 +58,9 @@ public interface Application : GLib.Application {
         shutdown.connect(() => {
             stream_interactor.connection_manager.make_offline_all();
+#if _WIN32
+            Dino.Util.CoUninitialize();
```
According to documentation:
> A thread must call CoUninitialize once for each successful call it has made to the CoInitialize or CoInitializeEx function, including any call that returns S_FALSE.

This is only once per application start.

LAGonauta replied:
Uhm.. Here it says that every call to `CoInitialize` must be balanced with a call to `CoUninitialize`:
https://docs.microsoft.com/pt-br/windows/win32/api/objbase/nf-objbase-coinitialize

Those COM stuff is confusing. I changed the code to make it call `CoUninitialize` as soon as possible after calling `CoInitialize`, not sure if this is correct...
Maybe I guess I could just remove this `CoInitialize` code, not sure if there is much use case to open something with components that require COM (such as shell extensions).

---

How many threads there is in Dino? Is the thread that starts/shutdowns Dino the same thread that calls ShellExecute?

mar-v-in replied:
Dino has multiple threads, but most of it happens in a single main thread. Start and shutdown should be in the same thread. But I guess your change is sufficient, even though I am not an expert on Windows things.

mar-v-in commented on main/src/ui/main_window.vala:
```diff
@@ -55,6 +55,17 @@ public class MainWindow : Gtk.Window {
     }
 
     private void setup_unified() {
+#if _WIN32 // WIN32 seems to not support exporting these symbols
```
This shouldn't be needed and is a sign that the build is not correct. I am very certain that exporting symbols is possible, if it's not, you might be using the wrong linker.

LAGonauta replied:
Done, I forgot to prefix the linker option with `-Wl`... that is why it wasn't working 🙄

mar-v-in commented on plugins/gpgme-vala/src/gpgme_helper.vala:
```diff
@@ -175,6 +175,9 @@ private static uint8[] get_uint8_from_data(Data data) {
 private static void initialize() {
     if (!initialized) {
+#if _WIN32
+        set_global_flag("w32-inst-dir", "C://Program Files (x86)//GnuPG//bin");
```
A hard-coded path won't work if the installation differs from what you expect. If it isn't reasonably possible to automatically find the correct path on all systems, we'd need to allow users to enter their path manually.

LAGonauta replied:
Uhm... what is the best way to handle this? Get the path from an environment variable, or add a textbox to the account config screen when OpenPGP is enabled?

mar-v-in replied:
Do gpg4win installations set some environment variable? I've never used it, so I don't know what's the way to go. What do other apps do?

LAGonauta replied:
Done

mar-v-in commented on main/dino-info.rc:
```diff
@@ -0,0 +1,25 @@
+1 VERSIONINFO
+FILEVERSION     0,1,0,0
```
Various versions in this file are hardcoded, but they should be set at build time.

LAGonauta replied:
Done

mar-v-in reviewed (commented):
Instead of using ifdef's, please move win32 specific code in a dedicated plugin that is only enabled by default on win32 platforms. This should be feasible at least for the CSS/font related change.

LAGonauta committed:
Set Windows executable version from PROJECT_VERSION

LAGonauta committed:
Add WIN32 fonts as a plugin

LAGonauta committed:
Every call to CoInitialize() must be balanced
with a call to CoUninitialize()

LAGonauta committed:
Add --export-all-symbols to Windows compilation

LAGonauta committed:
Add implicit link directories to package HINT path on MingW

Instead of blacklisting those libraries

LAGonauta committed:
Do not hardcode GPG path on Windows

linkmauve commented on libdino/src/service/util.vala:
```diff
@@ -26,6 +45,33 @@ public class Util {
+    public static void launch_default_for_uri(string file_uri)
+    {
+#if _WIN32
+        ShellExecute("open", file_uri);
```
Any reason this fallback is done in Dino instead of Glib?

LAGonauta replied:
Even though this works fine for Dino, it doesn't necessarily gives the control that GLib wants. See: https://gitlab.gnome.org/GNOME/glib/-/issues/1932#note_649492

This was also asked here: https://gitlab.gnome.org/GNOME/glib/-/issues/2088
With no direct answer. I can try to contact them on IRC to understand the issue completely, if you want.

LAGonauta committed:
Export all plugin symbols on Windows

LAGonauta committed:
Use Dino.Util.get_content_type also on preview

LAGonauta commented:
Rebased to master and added two new commits:
- Export all plugins symbols on WIN32, it is necessary now. Maybe a change in newest GCC?
- Images preview now also works on Windows

Deluvi commented:
I just wanted to thank you for your recent fixes on the plugins symbols ! The Windows build works like a charm 👍

LAGonauta committed:
Initial notification support

LAGonauta committed:
Refactor windows-notification plugin

LAGonauta committed:
Clean up

LAGonauta committed:
Use code from Dino.Ui.Util

LAGonauta committed:
Convert C code to Vala

LAGonauta committed:
Add callback support

LAGonauta committed:
Allow null image_path

LAGonauta committed:
Use dynamic linking instead of runtime loading

Also made me notice that the signature of the function with the callback was wrong. Oops.

LAGonauta committed:
Allow 32-bit linking

Win32 apis are __stdcall

LAGonauta commented:
Rebased to master and added Windows 32-bit compilation support
Was win32 apis use __stdcall, so I was missing those special windows headers.

LAGonauta committed:
Added 32-bit wintoast linker library

LAGonauta committed:
Use VAPI and generate template in-app

LAGonauta committed:
Initial plugin using new notification provider

LAGonauta committed:
Add support for custom actions on notification

LAGonauta committed:
Add notification retraction

LAGonauta committed:
Use list with all notifications

LAGonauta committed:
Rename field

LAGonauta committed:
Fix muc invite and voide request not working

LAGonauta committed:
Do not use GLib to open links in messages

Use ShellExecute

LAGonauta committed:
Use last_index_of instead of index_of

LAGonauta committed:
Initial callback support

LAGonauta committed:
Initial GObject wrapper for WinRT notifications

Still missing a lot of stuff

LAGonauta committed:
Initial code to allow buttons and text

LAGonauta committed:
Use string_view

LAGonauta committed:
Increase ref on event token

LAGonauta committed:
Add toastnotifier

LAGonauta committed:
Fix string conversion

LAGonauta committed:
Actions can stack

LAGonauta committed:
Fields must be created in the private struct

Also change unordered_map to list, we do not need hashing and stuff.

LAGonauta committed:
Add failed and dimissed actions

LAGonauta committed:
Cleanup dismissed actions on toast notification finalizer

LAGonauta committed:
Add template type enum

LAGonauta committed:
Rename enums to better match what Vala expects

LAGonauta committed:
Rename plugin vala file

LAGonauta committed:
Add template getter

LAGonauta committed:
Initial experiments with notification XML building

LAGonauta committed:
Anitial builder

LAGonauta committed:
Initial notification provider using WinRT

Crashes when activating actions, might be related to threads.

LAGonauta committed:
Delegate `activate_action` to UI thread

LAGonauta committed:
Fixed crash with multiple notifications

Sometimes an invalid function pointer was called with an invalid context

LAGonauta committed:
Add comment to builder

LAGonauta committed:
Use async

LAGonauta committed:
Use g_new0 and g_free to generate raw strings

LAGonauta committed:
Valac think that getters are always owned by the struct

Yuubi-san committed:
introduce try_invoke -- a logging exception catcher

Yuubi-san committed:
stop exceptions from crossing ABI boundary in a few places

Yuubi-san committed:
mark exception-safe C entry points as such

Yuubi-san committed:
clarify some entry points' names

Yuubi-san committed:
make GetCurrentModulePath and GetShortcutPath throw win32 errors

Yuubi-san committed:
clarify GetCurrentModulePath's name

Yuubi-san committed:
generalize GetShortcutPath into GetEnv

Yuubi-san committed:
make GetEnv more robust and not limit length of variables

Yuubi-san committed:
change some local functions' signatures

Yuubi-san committed:
constify all the things

Yuubi-san committed:
rewrite shortcut management code with RAII, error logging and exceptions

It actually works now.

Yuubi-san committed:
add restoration of shortcut's target path

Yuubi-san committed:
switch to runtime loading of PropVariantToStringAlloc

Now it really should work.

LAGonauta committed:
Add ginvoke to CMakeLists

LAGonauta committed:
Removed unused library on linker

It is loaded dynamically

LAGonauta committed:
Add README.md to Windows notification plugin

LAGonauta committed:
Fix notifications not hiding

Yuubi-san committed:
unimplement accidentally implemented wide string overloads of describe_argument

Yuubi-san committed:
work around GetEnvironmentVariable not resetting last error

Yuubi-san committed:
handle exe paths longer than 259 chars

Yuubi-san committed:
move some whitespace around

Yuubi-san committed:
use lower-case 0x prefix for hresult code formatting everywhere

Yuubi-san committed:
remove an unused include

Yuubi-san committed:
make meta-error messages more precise

Yuubi-san committed:
handle empty hresult_error message specially

Yuubi-san committed:
handle theoretical future failures of wsview_to_char

Yuubi-san committed:
fix UB in glib::describe_arguments called with no arguments

Makes failure logging of nullary invokables non-crashy.

Yuubi-san committed:
make glib::impl::varstring less explosive

Yuubi-san committed:
fiddle with punctuation

Yuubi-san committed:
add nullary version of g_try_invoke macro

Yuubi-san committed:
generalize glib::try_invoke to any return-by-value type and void

Yuubi-san committed:
protect GetTemplateContent callers from exceptions

Yuubi-san committed:
rewrite InitApartment and protect callers from (the rest of the) exceptions

Initializing COM by calling `winrt::init_apartment()` would always cause
stack unwinding *in practice*, which is suboptimal at best, and even using
`apartment_type::single_threaded` still would require exception filtering
*just in case*.

Yuubi-san committed:
handle empty menu-relative shortcut paths

Yuubi-san committed:
move module loading functions out of shortcutcreator.cpp

Yuubi-san committed:
work around a (pedantic) format specifier warning

Yuubi-san committed:
silence enum stringification warnings by first casting to underlying types

Yuubi-san committed:
fix / work around uninitialized fields warnings

Yuubi-san committed:
don't use FALSE as a null pointer constant

Yuubi-san committed:
replace C-style concurrent initialization of statics

C++ statics are thread-safe as is and are usually implemented more
efficiently. Besides, `volatile` is likely misused here anyway.

Yuubi-san committed:
reflow/respace

Yuubi-san committed:
stop checking for empty AUMIDs

The downstream code handles them just fine.

Yuubi-san committed:
log SetCurrentProcessExplicitAppUserModelID errors

Yuubi-san committed:
remove the no-longer-needed -municode compile option

Yuubi-san committed:
replace lists with vectors

Yuubi-san committed:
init `Callback` completely always

The `token` pointer was left dangerously uninitialized after construction.

Yuubi-san committed:
comment out unused arguments [-Wunused-parameter]

LAGonauta committed:
Add support for adaptive Windows 10 notifications

LAGonauta committed:
Add support for inline images to notification

LAGonauta committed:
Allow null header, body, applogo, and image on notification builder

LAGonauta committed:
Remove unity compilation unit

LAGonauta committed:
No need to enable coroutines

LAGonauta committed:
Initial code for using winrt headers

LAGonauta committed:
Add MIT licensed winrt headers

LAGonauta commented:
@fiaxh @mar-v-in 
Just pushed our notification work to this PR. Part of this PR are cppwinrt headers (licensed to MIT and distributed together with Windows SDK) modified by @Yuubi-san to compile on GCC. There is also a tool to generate those headers on the fly, however we did not manage to determine to which license they would be licensed yet.

However, I also noticed that Windows 8.1 is unable to load the plugin if Windows 8.1 SDK is not installed and those headers are used (work fine with the headers generated by the tool), even if the newest vcredist is installed. No solution has been found yet.

I chose to update this PR anyway to get feedback on anything else I should change.

LAGonauta commented:
![5d285e15-2dc7-4a5e-a77a-6e40a0b8fc28](https://user-images.githubusercontent.com/14236067/115160046-84365f00-a06c-11eb-9826-ebc23c40f21d.png)

LAGonauta committed:
Prefer primary DirectSound device on Windows

It automatically selects the default device for use,
there is no book keeping necessary and things just work

The primary DirectSound device has a (NULL) guid, making
it wasy to be found.

LAGonauta committed:
Do not allow selection of WASAPI devices

Dino would have to resample it own audio, do more book keeping and
somehow find out manually which is the default device.

LAGonauta committed:
Add initial call notifications

LAGonauta committed:
DelegateToUi must be an owned function

LAGonauta commented:
We found out what is causing Dino to not load the plugin on Windows 8.1, the linkage with the `mincore` library.
However, that library is not necessary when those WinRT headers are generated on-the-fly. We asked in `cppwinrt` repository and they think the generated code is licensed as MIT, but their answer did not go through their legal team:
https://github.com/microsoft/cppwinrt/issues/929

We are evaluating our options, but we would also like to know what you prefer. Some are:
1. Use headers generated by ourselves using YoloRT and commit them to the repository. Easy to use, trivial CI/CD and works everywhere. Fuzzy license state.
2. Use those explicitly MIT-licensed files from Windows SDK. Just as easy as last option, but does not work on Windows 8.1.
3. Generate the headers on compilation. Nothing is commited to Dino's repository, but the Windows binary might be tainted with code that might be (or not be) proprietary.

The perfect option is number 1, but might be risky. Option 2 is what we currently have, should be fine after January 10, 2023 (when official extended support for Windows 8.1 ends)
If we assume APIs are not copyrightable, number 3 might be our best option. If those APIs are ever ruled to be copyrightable and for some reason they are not MIT, it would be easy to just stop distributing binaries with the notification plugin built-in.

If Windows 8.1 support is not a priority, we can just use Option 2 (what we currently do)

Yuubi-san commented:
There's also the option
3a. Download pre-generated headers on build/configure time from... somewhere. The advantage being independence from Windows as a build host (Dino for Windows built fine on, e.g., Fedora, last time @LAGonauta checked). Of course, somebody somewhere still needs to generate the headers using windows metadata, but only once (or any time there are some new shiny features that Dino would like to use in, e.g., notification code).

Both option 3 and 3a don't create potential legal problems inside Dino's repo. 3 has the downside of requiring windows (license) to build the notification plugin; 3a suffers if/when the download location(s) go(es) down (like, microsoft suspending a github repo or its owner).

I'd also like to opine against dropping 8.1 support before *at least* MS officially stops patching its vulnerabilities. The mincore dependency is unnecessary bloat which is present only in the 1.5-year old version of the headers (the one from the SDK).

Yuubi-san commented:
Personally, I'm leaning towards option 1, but to be on the safe/pragmatic side, some variation of 3 is more preferable.

mar-v-in commented:
I would definitely prefer not having a bunch of header files that are not directly related to Dino in our source repository. If they can be generated at build time, that would be one option. Alternatively, requiring those header files to be around somewhere at build time would be fine as well.

In a perfect world those headers would be in a package that is discovered via pkg-config, but of course having an option to cmake would also be fine.
For CI builds we can also fetch them from somewhere, for release builds I wouldn't mind if those headers are only "assume"d to be MIT licensed - after all, this use is completely in the interest of the copyright holder anyways.

Yuubi-san commented:
> for release builds I wouldn't mind if those headers are only "assume"d to be MIT licensed - after all, this use is completely in the interest of the copyright holder anyways.
Yeah, definitely not a legal issue, only potentially a matter of ideological purity of the builds.

LAGonauta commented:
Just rebased and added some changes for proper A/V on Windows.

On Windows 10 it is possible to have extremely low audio latency, for example:
```bash
 gst-launch-1.0 -v wasapisrc ! audioamplify amplification=20.0 clipping-method=wrap-positive ! audioresample ! wasapisink low-latency=true
```

But I couldn't reproduce that in code yet, therefore Dino will just use DirectSound. Latency is comparable to Google Meet, so I think it is fine. With WASAPI latency should be similar to Mumble.

LAGonauta committed:
Use correct generic type for ArrayList

Nullable crashes Dino

LAGonauta committed:
Allow devices with properties and use has_classes

LAGonauta committed:
Remove YoloRT from tree

LAGonauta committed:
Build YoloRT on project build

LAGonauta committed:
Do not generate WinRT headers, just download them on build

danielreuther committed:
Fix compilation on gcc 11

Yuubi-san committed:
define _POSIX_C_SOURCE=1 on windows

Fixes "undefined reference to `localtime_r`" in, e.g., Vala's GLib.Time.local
when building on mingw-w64.

LAGonauta committed:
Merge pull request #5 from danielreuther/master-windows-changes-gcc11-fix

Fix compilation on GCC 11

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

# Conflicts:
#	main/src/ui/conversation_content_view/file_widget.vala
#	plugins/CMakeLists.txt
#	plugins/rtp/src/device.vala
#	plugins/rtp/src/plugin.vala

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
fix call notifications buttons not working

LAGonauta committed:
no need to ignore wasapi

LAGonauta committed:
Ignore wasapi devices as they do not work well yet

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

wurstsalat3000 commented:
@LAGonauta I'm not sure whether colored emoji rendering on Windows already works for you in these builds. Since [1.17.6](https://gitlab.freedesktop.org/cairo/cairo/-/tags/1.17.6), cairo supports rendering of fonts with colored emojis on Windows. I just figured out how to enable this for Gajim, and I thought I'd let you know:

Set the `PANGOCAIRO_BACKEND` environment variable to `fontconfig` (default is `win32`).

Yuubi-san commented:
Okay, now I'm hyped too... But no windows around to try it out :D

LAGonauta committed:
:Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Removed version from Dino executable

We need a better way to get the version number

LAGonauta committed:
Automatically set PANGOCAIRO_BACKEND to fontconfig on win32

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Fixed using GTK3 instead of GTK4

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Check YoloRT checksum before building

driver1998 commented:
C++/WinRT Upstream is working on improving MinGW support, and should already work with GCC 12 and LLVM 15.
I don't think YoloRT would be necessary anymore.

see:
https://github.com/mstorsjo/llvm-mingw/issues/307
https://github.com/alvinhochun/mingw-w64-cppwinrt

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Fix GPGME

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
Initial code to update Dino for Windows

YoloRT is not necessary anymore as msys2 has a wincpp-winrt package now
Needs more changes to compile due to webrtc audio processing

LAGonauta committed:
Removed yolort

LAGonauta committed:
Initial code to compile through meson

LAGonauta committed:
Enable build windows-notification through meson

LAGonauta committed:
Compoile dino rc

LAGonauta committed:
Initial windows CI code

LAGonauta committed:
rust first command on powershell

LAGonauta committed:
Add audio processing

LAGonauta committed:
Build for ARM

LAGonauta committed:
disable nsis for arm64

LAGonauta committed:
Build and upload dino for windows artifact

LAGonauta committed:
Use equal

LAGonauta committed:
Use upload v4

LAGonauta committed:
comment out installer

LAGonauta committed:
Properly check matrix value

LAGonauta committed:
Cannot compile to ARM yet

LAGonauta committed:
Use matrixed artifact upload name

LAGonauta committed:
Fixup

LAGonauta committed:
Export all symbols with meson

LAGonauta committed:
REVERT: Disable test temporarily

LAGonauta committed:
Properly set _WIN32

LAGonauta committed:
Install libsignal-protocol-c for OMEMO

LAGonauta committed:
Improve description

LAGonauta committed:
Organize definitions

LAGonauta commented:
Hello folks!
I updated my branch to latest Dino, added Meson build support, and added a build pipeline to build Dino though both GCC and Clang.

Some problems:
1. webrtc-audio-processing-1 is not being detected in the build pipeline for some reason, but it is installed. It is detected correctly locally, though... This still needs investigation.
2. glib-2.0 try_parse does not return the default value when parsing invalid numbers, which means one of the Stanza tests fails. Still needs investigation. As I wrote in the MUC: `On Windows, uint.try_parse of -42 returns 4294967254 instead of the default value 3... so it overflows instead of failing to parse`

I've been daily driving the GCC CMake build for some time now and it seems to be working fine, I should daily drive the meson build now.

LAGonauta committed:
Improve emoji rendering on Windows

LAGonauta committed:
Also build Windows with CMake

LAGonauta committed:
Re-enable test

LAGonauta committed:
Revert "Re-enable test"

This reverts commit 7f95d27c0095e804a6c817a4ea36dc6347b4b527.

LAGonauta committed:
Install webrtc-audio-processing v0.3

LAGonauta committed:
Install audio processing 0.3

LAGonauta committed:
Use global link arg

LAGonauta committed:
Delete anything that is not a dll from plugins dir

LAGonauta committed:
Add global link and build release

LAGonauta committed:
Fix package script

LAGonauta committed:
disable stripping cmake

LAGonauta committed:
Fix webrtc audio processing and restore project link arg

LAGonauta committed:
Update cmake version

LAGonauta committed:
Move plugins dir to be beside dino executable

LAGonauta committed:
Delete what is not a plugin

LAGonauta committed:
Move plugins to proper folder

LAGonauta committed:
Fixup plugin search

LAGonauta committed:
Use make install instead

LAGonauta committed:
Do the same for meson

LAGonauta committed:
Copy meson plugins to the proper folder

LAGonauta committed:
Fix emoji, hopefully

Neustradamus commented:
To follow this good @LAGonauta PR :)

LAGonauta committed:
Fix fonts plugin and command prompt

LAGonauta committed:
Set win_subsystem properly

LAGonauta committed:
Test without exporting all symbols

LAGonauta committed:
Revert "Test without exporting all symbols"

This reverts commit 6d6864749d1c66fecab0afe5f51daa1105472740.

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes-test-ci

LAGonauta committed:
Merge branch 'master-windows-changes-test-ci' into master-windows-changes

LAGonauta committed:
Remove extra CMakeLists from Windows plugins

LAGonauta committed:
Merge remote-tracking branch 'upstream/master' into master-windows-changes

LAGonauta committed:
No need to use ShellExecute anymore
